mantle/kola/tests/ignition/qemufailure.go

```diff
@@ -101,3 +129,206 @@ func ignitionFailure(c cluster.TestCluster) error {
+			return err
+		}
+		return nil
+	}
+}
+
+// Use ignition config to create a second bootfs
+// 1 - produce an ignition file that format a disk with a"boot" label.
+// 2 - boot the VM with the ignition file and an extra disk.
+// 3 - observe the failure 
+func dualBootfsIgnitionFailure(c cluster.TestCluster) error {
+	builder := platform.NewQemuBuilder()
+	// inserting log file for troubleshooting
+	// file located in coreos-assembler directory
+
+	// consoleFile, err := builder.TempFile("console.log")
+	// if err != nil {
+	// 	return err
+	// }
+	//builder.ConsoleFile = consoleFile.Name()
+	builder.ConsoleFile = "console.txt"
+
+	failConfig, err := conf.EmptyIgnition().Render(conf.FailWarnings)
+	if err != nil {
+		return errors.Wrapf(err, "creating empty config")
+	}
+
+	// Craft an ingniton file that format a partition
+	formaterConfig := types.Config { 
+		Ignition: types.Ignition{
+			Version: "3.2.0",
+		},
+		Storage: types.Storage { 
+			Filesystems: []types.Filesystem{
+				{
+					Device:         "/dev/disk/by-id/virtio-extra-boot",
+					Label:          util.StrToPtr("boot"),
+					Format:         util.StrToPtr("vfat"),
+					WipeFilesystem: util.BoolToPtr(true),
+				},
+			},
+		},
+	}
+	failConfig.MergeV32(formaterConfig)
+
+	builder.SetConfig(failConfig)
+	err = builder.AddBootDisk(&platform.Disk{
+		BackingFile: kola.QEMUOptions.DiskImage,
+	})
+
+	if err != nil {
+		return err
+	}
+
+	err = builder.AddDisksFromSpecs([]string{"1G:serial=extra-boot"})
+	if err != nil {
+		return err
+	}
+
+	builder.MemoryMiB = 1024
+	builder.Firmware = kola.QEMUOptions.Firmware
+	inst, err := builder.Exec()
+	if err != nil {
+		return err
+	}
+	defer inst.Destroy()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Minute)
+	defer cancel()
+
+	errchan := make(chan error)
+	go func() {
+		resultingError := inst.WaitAll(ctx)
+
+		if resultingError == nil {
+			resultingError = fmt.Errorf("bootfs unexpectedly succeeded")
+		} else if resultingError == platform.ErrInitramfsEmergency {
+			// Expectred initramfs failure, checking the console file to insure
+			// that coreos-unique-boot.service failed
+			b, err := os.ReadFile(builder.ConsoleFile)
+			if err != nil {
+				panic(err)
+			}
+			isExist, err := regexp.Match("Error: System has 2 devices with a filesystem labeled 'boot'", b)
+			if err != nil {
+				panic(err)
+			}
+			if isExist {
+				// The expected case
+				resultingError = nil
+			} else {
+				resultingError = errors.Wrapf(err, "expected coreos-unique-boot.service to fail")
+			}
+		} else {
+			resultingError = errors.Wrapf(err, "expected initramfs emergency.target error")
+		}
+		errchan <- resultingError
+	}()
+
+	select {
+	case <-ctx.Done():
+		if err := inst.Kill(); err != nil {
+			return errors.Wrapf(err, "failed to kill the vm instance")
+		}
+		return errors.Wrapf(ctx.Err(), "timed out waiting for initramfs error")
+	case err := <-errchan:
+		if err != nil {
+			return err
+		}
+		return nil
+	}
```

Comment:
So now we have roughly three copies of this. Probably time to factor it out into its own function and have all three tests call out to it. The function can take the string to grep for in the console logs.

Reply:
It took some time, but I think it now works as expected.

Reply:
I'm wondering now if that's what you meant.